tests/_internal/config/test_assemble_build_config.py

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+import json
+import contextlib
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from unittest import TestCase
+
+from mbed_build._internal.config.assemble_build_config import _assemble_config
+from mbed_build._internal.config.config import Config
+from mbed_build._internal.find_files import find_files
+from mbed_build._internal.config.source import Source
+from tests._internal.config.factories import SourceFactory
+
+
+@contextlib.contextmanager
+def create_files(files):
```

Comment:
Is it worth pulling this out into a `factory.py` or something? I can see this being very useful in other tests in the future.

Reply:
Hm. This is json specific and I don't see how that'd be reusable for other tests right now.

I tried extracting generic file creation, but that'd mean wrapping all the dicts in `json.dumps` or creating a local `contextmanager` that does `with_create_json_files`, that uses this shared `with_create_files`.

Tempted to leave it as is for now. Too many decisions.